implementation/src/main/java/io/smallrye/config/SmallRyeConfigMappingProvider.java

```diff
@@ -0,0 +1,38 @@
+package io.smallrye.config;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import io.smallrye.config.mapper.ConfigMappingProvider;
+import io.smallrye.config.mapper.ConfigurationObject;
+import io.smallrye.config.mapper.ConfigurationValidationException;
+
+public class SmallRyeConfigMappingProvider implements Serializable {
```

Comment:
This class probably should not exist.  The `ConfigMappingProvider` does not need to exist once the corresponding configuration is mapped; the roots should then be associated directly with the configuration.  Otherwise this holds on to a potentially significant amount of memory for an indefinite amount of time.

Also calling this class `SmallRyeConfigMappingProvider` when it does not extend `ConfigMappingProvider` is somewhat confusing.

Reply:
I was thinking about that (I'm terrible for naming :D).